R/fit_model.R

```diff
@@ -160,15 +160,24 @@ format_right_truncation <- function(pmf, data) {
     # silently removed if length of the PMF was longer than the data,
     # effectively eliminating the right-truncation correction
 
-    cli::cli_abort(
+    trunc_idx <- nrow(data) - 1
+    cli::cli_warn(
       c(
+        "Removing right-truncation PMF elements after {.val {trunc_idx}}",
         "Right truncation PMF longer than the data",
         "PMF length: {.val {length(pmf)}}",
         "Data length: {.val {nrow(data)}}",
-        "PMF can only be up to length as the data"
+        "PMF can only be up to the length of the data"
       ),
       class = "right_trunc_too_long"
     )
+    suppressWarnings({
+      EpiNow2::trunc_opts(
```

Comment:
Does this renormalise it?

Reply:
As in I assume it's bad to just truncate the PMF if the data is too short and not renormalise

Reply:
It doesn't renormalize but that was on purpose. It's really about the meaning of the last delay being at, say, 75%. We'd see latent nowcasted cases be at 133% of the final observed value and we'd be able to compare nowcasted cases relative to final observations.

EpiNow2 doesn't check for PMF validity of the right-truncation PMF, so this shouldn't be an operational concern.

@seabbs can you confirm the above?

Reply:
Resolving per Sam's confirmation.